mteb/models/nvidia_models.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+from functools import partial
+from typing import Any
+
+import numpy as np
+import torch
+from sentence_transformers import CrossEncoder, SentenceTransformer
+
+from mteb.encoder_interface import PromptType
+from mteb.model_meta import ModelMeta
+from mteb.models.sentence_transformer_wrapper import SentenceTransformerWrapper
+
+logger = logging.getLogger(__name__)
+
+
+def instruction_template(instruction: str) -> str:
+    return f"Instruct: {instruction}\nQuery: " if instruction else ""
+
+
+class NvEmbedWrapper(SentenceTransformerWrapper):
```

Comment:
Why not use gritlm?

Reply:
Because nvidia uses custom code to get embedding, `model.encode(...)` instead of `model(...)`. This is not compatible with gritlm without making changes in `grit` library

Reply:
GritLM uses `model.encode(...)` too. @Muennighoff can you help please?

Reply:
Yeah I think it should also work with the [InstructWrapper](https://github.com/embeddings-benchmark/mteb/blob/19aefa36e7b7aae74065dbe8c78bed8c1716b6e5/mteb/models/instruct_wrapper.py#L30). One reason we like using the InstructWrapper, is that it can use multiple GPUs while SentenceTransformer can only use 1 GPU afaik, but either way works for me


also cc @YashDThapliyal to let you know that NVEmbed is already being implemented so maybe you want to focus on a different model instead e.g. https://huggingface.co/jxm/cde-small-v1

Reply:
FYI `cde` have mteb evaluation scripts https://github.com/jxmorris12/cde/blob/bcf31db980dd9e7864cc635333f79f405fd5de03/evaluate_mteb.py, but it is very unusual

Reply:
@Muennighoff 
In [this line](https://github.com/ContextualAI/gritlm/blob/e59fd971cb4b148d1d59130aa5a0a7a76de6e320/gritlm/gritlm.py#L134C66-L134C80) we have to call `model.encode(..)`. This can be done if `self.embedding_attr == ‘encode’`. But for embedding mode `self.embedding_attr` is [always None](https://github.com/ContextualAI/gritlm/blob/e59fd971cb4b148d1d59130aa5a0a7a76de6e320/gritlm/gritlm.py#L31). This means you are calling `model(..)`, which gives incorrect embeddings. Are you suggesting to change the code in the `gritlm` library to fix this?

Reply:
Oh I see, sorry for missing this! Maybe let's leave it as is with your sentence transformers implementation then for simplicity!

(as you say we could probably modify gritlm or copy over the https://huggingface.co/nvidia/NV-Embed-v2/blob/main/modeling_nvembed.py#L416 bits such that just forward can be called but maybe this adds too much complexity for now)